src/convertToObject.js

```diff
@@ -6,7 +6,15 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = sourceString.split(';').map((pair) => {
+    const [key, value] = pair.trim().split(':');
+
+    if (key && value) {
+      return { [key.trim()]: value.trim() };
+    }
+  });
+
+  return Object.assign({}, ...result);
```

Comment:
The use of 'Object.assign' here is correct, but ensure that the variable name 'result' is more descriptive as per the checklist.